src/lsm/schema.zig

```diff
@@ -58,6 +58,14 @@ const block_body_size = block_size - @sizeOf(vsr.Header);
 const BlockPtr = *align(constants.sector_size) [block_size]u8;
 const BlockPtrConst = *align(constants.sector_size) const [block_size]u8;
 
+pub inline fn header_from_block(block: anytype) switch (@TypeOf(block)) {
+    BlockPtr => *vsr.Header,
+    BlockPtrConst => *const vsr.Header,
+    else => unreachable,
+} {
+    return mem.bytesAsValue(vsr.Header, block[0..@sizeOf(vsr.Header)]);
+}
```

Comment:
Instead of two different helpers, I restored the use cases where the block can be invalid with `mem.bytesAsValue`,
and now the helper signature is `*const` only, which prevents misuse.

I did that mostly because I couldn't find good names for splitting the function into two, and I realized that `mem.bytesAsValue` is in fact great for what we are doing in those places.
Suggestions are accepted 😁